Persephone/Components/Browser/Album Detail/AlbumDetailView.swift

```diff
@@ -113,15 +113,18 @@ class AlbumDetailView: NSViewController {
   }
 
   func getAlbumSongs(for album: Album) {
-    App.mpdClient.getAlbumSongs(for: album.mpdAlbum) { [self] (mpdSongs: [MPDClient.MPDSong]) in
-      self.dataSource.setAlbumSongs(
-        mpdSongs.map { Song(mpdSong: $0) }
-      )
+    App.mpdClient.getAlbumSongs(for: album.mpdAlbum) { [weak self] (mpdSongs: [MPDClient.MPDSong]) in
+      guard let self = self else { return }
 
       DispatchQueue.main.async {
+        self.dataSource.setAlbumSongs(
+          mpdSongs.map { Song(mpdSong: $0) }
+        )
+
         self.albumTracksView.reloadData()
 
-        guard let song = self.dataSource.albumSongs.first?.song ??
+        guard !self.dataSource.albumSongs.isEmpty,
+          let song = self.dataSource.albumSongs.first?.song ??
           self.dataSource.albumSongs[1].song
```

Comment:
Wait, if we're in this block because `self.dataSource.albumSongs.isEmpty`, then isn't
`self.dataSource.albumSongs[1].song` going to crash?

Reply:
Looking at the code more deeply, if `self.dataSource.albumSongs.first?.song` is nil, isn't the `self.dataSource.albumSongs[1].song` alternative going to crash?

Reply:
So this is a somewhat weird bit of code. The first item in the `albumSongs` array will be either a song (in the case of a single disc album) or a disc heading (for multidisc albums). Therefore, if the first item contains a nil song then it's the second item that contains the first song of the album.

I actually store the first song of the album as part of the album struct now so I can refactor this to use that instead.